### PR TITLE
New env design, more docs, misc updates

### LINKED ISSUE #36: Questions service constants file
Currently, the defaults for our env variables are spread across various files.

Let's combine them into a single constants file, and update the docs accordingly.

### LINKED ISSUE #44: Standardise which env files are committed
Frontend currently has an env file that is committed. The design seems inverted for our two services, where they should not be committed. There may be room to discuss a more optimal design.

### LINKED ISSUE #49: Update user docs that password is no longer included in responses
We have modified the provided user service such when returning user data, the hashed password is no longer returned. The existing documentation already goes over expected API inputs and outputs. We should update our documentation, such as the README file, to reflect our change.

### PR DESCRIPTION
**Pull request by Cloud7050**
Mostly closes #36. Closes #44, closes #49.

**Breaking Changes**

- Devs will need to update their env files accordingly
  - `.env`, which is already checked in, should cover most of the necessary config
  - Rename your `.env.development`s to `.env.local`. You should be able to delete most if not all of your local config. Refer to the various `.example` files for what they should now contain.\
  Or, refer to the updated READMEs to see which variables are required or optional

---

### General

- Standardise to new env design for all services.\
  We now mirror Vite's design. We essentially have `.env` checked in, `.env.local` not checked in, and `.env.development.local` if needed (also not checked in).\
  Refer to the READMEs for full details
- Tidy gitignores, dockerignore

### Common

- New `@common/utils` package to share util functions and constants between all services
  - Contains function to merge env files under the new design. All files should now be updated to this

### Frontend

- Clean up temp vite file

### User

- Remove jest and related edits

### Docs

- Update dev instructions to reflect D3. Include Docker, production steps
- Update obsolete info and move relevant details to the root README instead
- Document new env design, and update variable docs in services
- Document managing dependencies with `pnpm`
- Add WinNAT troubleshooting steps

### CODE REVIEW

**Comment by Cloud7050** [comment]
Opted for 4 space nested list indentation based on https://stackoverflow.com/a/53867805/11536796. Open to discussion.

**Comment by Cloud7050** [comment]
WinNAT troubleshooting instructions based on https://stackoverflow.com/a/69129708/11536796. See its linked GitHub issue comments for more context.

File: frontend/README.md [hunk 1]
```diff
@@ -1,32 +1,22 @@
...
-   ```
+# Frontend
+
+## Development Instructions
+
+Unless otherwise specified, commands should be run in the service root.
+
+### Running the service (development)
+
+1. Run `npm run dev` to run the dev script.
```
Comment (donljh) [inline frontend/README.md]: Should update to `pnpm` here

File: services/question/README.md [hunk 2]
```diff
@@ -1,51 +1,29 @@
...
-if (process.env.NODE_ENV) {
-  dotenv.config({ path: `.env.${process.env.NODE_ENV}` })
-} else {
-  dotenv.config({ path: '.env' })
-}
-```
+### Running the service (development)
 
-### Routes
+1. Run `npm run dev` to run the dev script.
```
Comment (donljh) [inline services/question/README.md]: Same here, and all the same instances below

File: README.md [hunk 3]
```diff
@@ -8,24 +8,102 @@ This is the monorepo for our PeerPrep implementation.
...
-1. Open a command line in the monorepo's root folder and run `pnpm i` to install all dependencies.
-   - You do not need to install dependencies individually within subfolders.
+1. Run `pnpm i` to install all dependencies.
+    - You do not need to install dependencies individually within subfolders.
+1. You would have been provided a JSON file containing Firebase service account credentials.\
+  Place it in `.firebase/service-account.json`.
+
+**Only** if the questions database has yet to be initialised and is thus empty:
+
+1. Navigate to the question service's root and run `npm run upload` to populate it.
```
Comment (donljh) [inline README.md]: Lets standardise to `pnpm`

File: README.md [hunk 4]
```diff
@@ -8,24 +8,102 @@ This is the monorepo for our PeerPrep implementation.
...
+1. Navigate to the user service's root and run `npm run create-admin` to create the account.
+
+#### For development
+
+1. Make a copy of `services/user/.env.local.example` and name it `services/user/.env.local`. Change the sample `JWT_SECRET` value for security reasons.
+
+#### For production
+
+1. Change the contents of `jwt_secret.txt` for security reasons.\
+  Do not check the new value in.
```
Comment (donljh) [inline README.md:49]: Sentence cut off here?

File: README.md [hunk 5]
```diff
@@ -8,24 +8,102 @@ This is the monorepo for our PeerPrep implementation.
...
 
 ### Running the project (development)
 
 1. Run `pnpm dev` to run the dev script.
+    - This starts the frontend, user service, and question service in dev mode.
+1. Open your browser at <http://localhost:3000>.
+
+### Running the project (production)
+
+1. Run `docker compose up`.
```
Comment (donljh) [inline README.md]: Might also want to change to `docker compose up -d` to run in a detached state (no need to open separate command prompt)

File: README.md (hunk 4 above)
Comment (Cloud7050) [reply README.md:49]: Ah not cut off - do not check xx in / do not check in xx, ie do not check the JWT secret in and push.
After #68 we'll need to update this part of the docs too.

**Comment by Cloud7050** [comment]
Note to self: ensure example files are dockerignored

Edit: `**/.env.*.example` taken care of by `**/.env.*`

**Review by donljh — APPROVED** [review]

**Review by ys112 — APPROVED** [review]

**Pull request merged**